esphome/components/online_image/online_image.cpp

```diff
@@ -100,7 +100,27 @@ void OnlineImage::update() {
   }
   ESP_LOGI(TAG, "Updating image %s", this->url_.c_str());
 
-  this->downloader_ = this->parent_->get(this->url_);
+  std::list<http_request::Header> headers = {};
+
+  http_request::Header accept_header;
+  accept_header.name = "Accept";
+  switch (this->format_) {
+    case ImageFormat::BMP:
+      accept_header.value = "image/bmp;q=0.9,*/*;q=0.8";
```

Comment:
Why explicitly limit the "main" format? Why not just
```suggestion
      accept_header.value = "image/bmp,*/*;q=0.8";
```
?

Reply:
I like it! I've updated this MR accordingly.